### PR TITLE
LPS-101582 If the bundle has a fragment, recompile jsps

### PR DESCRIPTION
**Pull request by mtambara**

### CODE REVIEW

**Comment by mtambara** [comment]
ci:test:relevant

File: modules/apps/static/portal-osgi-web/portal-osgi-web-servlet-jsp-compiler/src/main/java/com/liferay/portal/osgi/web/servlet/jsp/compiler/internal/JspServlet.java [hunk 1]
```diff
@@ -216,6 +221,40 @@ public void init(final ServletConfig servletConfig)
...
+				}
+
+				return bundle;
+			}
+
+		};
+
+		bundleTracker.open();
+
+		bundleTracker.close();
```
Comment (gamerson) [inline modules/apps/static/portal-osgi-web/portal-osgi-web-servlet-jsp-compiler/src/main/java/com/liferay/portal/osgi/web/servlet/jsp/compiler/internal/JspServlet.java]: hey @mtambara  Will this handle the case where initally when the JSP servlet is called, there is no attached fragments.  But then at some point later, the developer deploys a fragment.  Will the bundle tracker that sets the `hasFragment` be re-called?

Comment (mtambara) [reply modules/apps/static/portal-osgi-web/portal-osgi-web-servlet-jsp-compiler/src/main/java/com/liferay/portal/osgi/web/servlet/jsp/compiler/internal/JspServlet.java]: @gamerson Yes, when a fragment is attached the host bundle needs to be refreshed, creating a new JSP servlet and rechecking for fragment bundles

Comment (gamerson) [reply modules/apps/static/portal-osgi-web/portal-osgi-web-servlet-jsp-compiler/src/main/java/com/liferay/portal/osgi/web/servlet/jsp/compiler/internal/JspServlet.java]: Gotcha, makes sense.

**Comment by shuyangzhou** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/79305

**Pull request closed without merging**